### PR TITLE
Add support for "NaN" as a double value

### PR DESCRIPTION
**Pull request by rg3igalia**
This PR adds support for "NaN", "nan" and other lowercase/uppercase combinations to be interpreted as a double value token and read using strtod in the tokenizer.

This is interesting for an upcoming VK-GL-CTS CL I'm preparing in which I test some SPIR-V instructions like OpFOrdEqual or OpFUnordEqual. To make the new tests interesting I'd like to be able to pass NaN to amber and make sure it's properly taken into account for ordered and unordered instructions.

### CODE REVIEW

**Comment by rg3igalia** [comment]
I signed it!

File: src/tokenizer_test.cc [hunk 1]
```diff
@@ -77,6 +78,18 @@ TEST_F(TokenizerTest, ProcessDouble) {
   EXPECT_TRUE(next->IsEOS());
 }
 
+TEST_F(TokenizerTest, ProcessNaN) {
+  Tokenizer t("NaN");
```
Comment (dj2) [inline src/tokenizer_test.cc]: Can you make this a parameterized test and try different combinations? NaN, nan, NAN, etc?

File: src/tokenizer_test.cc [hunk 2]
```diff
@@ -77,6 +78,18 @@ TEST_F(TokenizerTest, ProcessDouble) {
...
+  auto next = t.NextToken();
+  ASSERT_TRUE(next != nullptr);
+  EXPECT_TRUE(next->IsDouble());
+  EXPECT_TRUE(std::isnan(next->AsDouble()) != 0);
```
Comment (dj2) [inline src/tokenizer_test.cc]: std::isnan returns a bool, so just EXPECT_TRUE(std::isnan(next->AsDouble()))

File: src/executor_test.cc [hunk 3]
```diff
@@ -202,7 +202,7 @@ class VkScriptExecutorTest : public testing::Test {
     auto engine = MakeUnique<EngineStub>();
     engine->Initialize(nullptr, nullptr, features, instance_extensions,
                        device_extensions);
-    return std::move(engine);
+    return engine;
```
[the remark this answers is not in the document]
Comment (rg3igalia) [reply src/executor_test.cc]: Yes, and moreover it appears some CI runs are failing after removing this, so I'm dropping the commit.

File: src/tokenizer_test.cc (hunk 1 above)
Comment (rg3igalia) [reply src/tokenizer_test.cc]: Will do.

File: src/tokenizer_test.cc (hunk 2 above)
Comment (rg3igalia) [reply src/tokenizer_test.cc]: Ah, you're right. isnan from C returns an integer but std::isnan returns a bool. Thanks!

**Pull request merged**